systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/AlternativeReconcileTriggersST.java

```diff
@@ -453,6 +458,178 @@ void testAddingAndRemovingJbodVolumes() {
+     * checking each configured volume for the presence of KRaft metadata files. The method asserts that the metadata log
+     * exists only in the specified KRaft metadata volume and not in others, ensuring correct metadata log placement
+     * according to the test configuration.
+     *
+     * @param testStorage               An instance of TestStorage containing configuration and context for the current test,
+     *                                  including namespace and broker selector for identifying relevant Kafka pods.
+     * @param kraftMetadataVolumeId     The volume ID expected to contain the KRaft metadata log. This method will
+     *                                  assert the presence of metadata logs in this volume and their absence in others.
+     * @param numberOfVolumes           The total number of volumes configured in the JBOD (Just a Bunch Of Disks) for each Kafka broker.
+     *                                  This dictates how many volume directories the method will check within each Kafka pod.
+     */
+    private void verifyKafkaKraftMetadataLog(final TestStorage testStorage, final int kraftMetadataVolumeId, final int numberOfVolumes) {
+        final List<Pod> kafkaPods = kubeClient().listPods(testStorage.getNamespaceName(), testStorage.getBrokerSelector());
+        int kafkaIndex = 0; // Ensure this index is managed appropriately if used outside this method context.
+
+        for (final Pod kafkaPod : kafkaPods) {
+            List<String> directories = new ArrayList<>();
+
+            // Dynamically create a list of volume directories to check, based on the total number of volumes.
+            for (int volumeId = 0; volumeId < numberOfVolumes; volumeId++) {
+                directories.add("/var/lib/kafka/data-" + volumeId + "/kafka-log" + kafkaIndex + "/__cluster_metadata-0");
+            }
+
+            // Check each directory in the current Kafka pod for the presence or absence of the metadata log.
+            for (final String dir : directories) {
+                final String result = cmdKubeClient().namespace(testStorage.getNamespaceName()).execInPodContainer(kafkaPod.getMetadata().getName(),
+                    "kafka",
+                    "/bin/bash", "-c", "ls " + dir + " && echo exists || echo not exists").out().trim();
+
+                LOGGER.info("Kafka pod: {} the directory: {} - {}", kafkaPod.getMetadata().getName(), dir, result);
+
+                // Assert the condition that metadata should only exist in the specified KRaft metadata volume.
+                if (dir.equals("/var/lib/kafka/data-" + kraftMetadataVolumeId + "/kafka-log" + kafkaIndex + "/__cluster_metadata-0")) {
+                    assertThat(result, CoreMatchers.containsString("exists"));
+                } else {
+                    assertThat(result, CoreMatchers.containsString("not exists"));
+                }
+            }
+            kafkaIndex++;
+        }
```

Comment:
I think that you don't have to fill some list of directories in one for loop and then iterate through it right after that. It is better to check it immediately, or?

Also, you don't have to do `ls` but `test -d` and then just instead of `out()` do `returnCode()` or?

Reply:
is the `test` utility part of the Kafka container? I am not sure about that

Reply:
isn't it the most basic thing on linux?

Reply:
```[kafka@47035107f40f kafka]$ test -d config
[kafka@47035107f40f kafka]$ echo $?
0
[kafka@47035107f40f kafka]$ test -d something
[kafka@47035107f40f kafka]$ echo $?
1
```

Reply:
Sure I can re-write it 👍.

Reply:
Update ;)